ED/src/init/ed_params.f90

```diff
@@ -2314,1771 +5945,936 @@ end subroutine init_pft_mort_params
-      b1Bs_small(    3)  = 0.21445616
-      b1Bs_small(    4)  = 0.26890751
-      b1Bs_small(14:16)  = 0.05291854
-      b1Bs_small(   17)  = 0.15940854
-
-      b2Bs_small(    1)  = 3.706955
-      b2Bs_small(    2)  = 2.342587
-      b2Bs_small(    3)  = 2.370640
-      b2Bs_small(    4)  = 2.254336
-      b2Bs_small(14:16)  = 3.706955
-      b2Bs_small(   17)  = 2.342587
-
-      b1Bl_large    (:)  = b1Bl_small(:)
-      b2Bl_large    (:)  = b2Bl_small(:)
-      b1Bs_large    (:)  = b1Bs_small(:)
-      b2Bs_large    (:)  = b2Bs_small(:)
-   end if
-   !---------------------------------------------------------------------------------------!
+   !----- Louis (1979) model. -------------------------------------------------------------!
+   bl79        = 5.0    ! b prime parameter
+   csm         = 7.5    ! C* for momentum (eqn. 20, not co2 char. scale)
+   csh         = 5.0    ! C* for heat (eqn.20, not co2 char. scale)
+   dl79        = 5.0    ! ???
+   !----- Oncley and Dudhia (1995) model. -------------------------------------------------!
+   beta_s       = 5.0          ! Beta
+   !----- Beljaars and Holtslag (1991) model. ---------------------------------------------!
+   abh91       = -1.00         ! -a from equation  (28) and (32)
+   bbh91       = -twothirds    ! -b from equation  (28) and (32)
+   cbh91       =  5.0          !  c from equations (28) and (32)
+   dbh91       =  0.35         !  d from equations (28) and (32)
+   ebh91       = -twothirds    ! - factor multiplying a*zeta in equation (32)
+   fbh91       =  1.50         ! exponent in equation (32)
+   cod         = cbh91/dbh91   ! c/d
+   bcod        = bbh91 * cod   ! b*c/d
+   fm1         = fbh91 - 1.0   ! f-1
+   ate         = abh91 * ebh91 ! a * e
+   atetf       = ate   * fbh91 ! a * e * f
+   !----- Similar to CLM (2004), but with different phi_m for very unstable case. ---------!
+   zetac_um    = -1.5
+   zetac_uh    = -0.5
+   zetac_sm    =  1.0
+   zetac_sh    =  zetac_sm
+   !----- Define chim and chih so the functions are continuous. ---------------------------!
+   chim        = (-zetac_um) ** onesixth / sqrt(sqrt(1.0 - gamm * zetac_um))
+   chih        = cbrt(-zetac_uh) / sqrt(1.0 - gamh * zetac_uh)
+   beta_vs     = 1.0 - (1.0 - beta_s) * zetac_sm
+   !----- Define derived values to speed up the code a little. ----------------------------!
+   zetac_umi   = 1.0 / zetac_um
+   zetac_uhi   = 1.0 / zetac_uh
+   zetac_smi   = 1.0 / zetac_sm
+   zetac_shi   = 1.0 / zetac_sh
+   zetac_umi16 = 1.0 / (- zetac_um) ** onesixth
+   zetac_uhi13 = 1.0 / cbrt(-zetac_uh)
 
-   !------------------------- Liana leaf Biomass (Putz, 1983) -----------------------------!
-   b1Bl_small(17) = b1Bl_small(2)
-   b2Bl_small(17) = b2Bl_small(2)
-   b1Bl_large(17) = 0.0856
-   b2Bl_large(17) = 2.0
-   !---------------------------------------------------------------------------------------!
 
+   !----- Parameters for the z0m:z0h ratio, following Zeng and Dickin
```

Comment:
This is a good place to define what `vis` and `nir` mean in terms of wavelength ranges. Otherwise, it's difficult to say how realistic these values actually are.

In my experience with the remote sensing literature, "visible" is about 400-700/750nm, "near-infrared" is from the end of visible to about 1400nm, 1400-2500 is "shortwave-infrared" (SWIR; which ED2 doesn't have), everything from there to like 5000nm is "mid-infrared" (MIR), and from there on is thermal.

@serbinsh @mrjohnston -- thoughts?

Reply:
Yep, pyrgeometers typically measure from 4.5 or 5 micrometers.